plugin/src/App/StatusPages/Settings/Setting.lua

```diff
@@ -53,22 +56,48 @@ function Setting:render()
 				return UDim2.new(1, 0, 0, 20 + value.Y + 20)
 			end),
 			LayoutOrder = self.props.layoutOrder,
+			ZIndex = -self.props.layoutOrder,
 			BackgroundTransparency = 1,
 
 			[Roact.Change.AbsoluteSize] = function(object)
 				self.setContainerSize(object.AbsoluteSize)
 			end,
 		}, {
-			Checkbox = e(Checkbox, {
-				active = self.state.setting,
+			Input = if self.props.options ~= nil then
+				e(Dropdown, {
+					options = self.props.options,
+					active = self.state.setting,
+					transparency = self.props.transparency,
+					position = UDim2.new(1, 0, 0.5, 0),
+					anchorPoint = Vector2.new(1, 0.5),
+					onClick = function(option)
+						Settings:set(self.props.id, option)
+					end,
+				})
+			else
+				e(Checkbox, {
+					active = self.state.setting,
+					transparency = self.props.transparency,
+					position = UDim2.new(1, 0, 0.5, 0),
+					anchorPoint = Vector2.new(1, 0.5),
+					onClick = function()
+						local currentValue = Settings:get(self.props.id)
+						Settings:set(self.props.id, not currentValue)
+					end,
+				}),
+
+			Reset = self.props.onReset and e(IconButton, {
+				icon = Assets.Images.Icons.Reset,
+				iconSize = 24,
+				color = theme.BackButtonColor,
 				transparency = self.props.transparency,
-				position = UDim2.new(1, 0, 0.5, 0),
-				anchorPoint = Vector2.new(1, 0.5),
-				onClick = function()
-					local currentValue = Settings:get(self.props.id)
-					Settings:set(self.props.id, not currentValue)
-				end,
-			}),
+				visible = self.props.showReset,
+
+				position = UDim2.new(1, -32 - (self.props.options ~= nil and 120 or 40), 0.5, 0),
```

Comment:
120 is a lot of padding between the text and the options I think 80 is better since then you have the minimum padding needed.
since if we where to try and fix #592 then the padding would become an issue

Reply:
Padding is there to support the reset button and its hover circle.
![image](https://user-images.githubusercontent.com/40185666/182273660-58f88522-61ed-4269-8392-94ff490c2755.png)

Reply:
2 Question 
why is the hover circle needed would it not be better to auto update the log level on change of the options?

the other one is you could probebly remove a little bit of padding still we could also do that once we fix the size issues later

Reply:
I think you misunderstand the point of that button. Of course we auto update the log level when you pick an option. When you have a non-default option selected, this button appears and clicking it resets the setting back to default for you.